src/lib/swapper/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts

```diff
@@ -0,0 +1,65 @@
+import type { Step } from '@lifi/sdk'
+import type { ChainId } from '@shapeshiftoss/caip'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import type { AbiItem } from 'web3-utils'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { SwapError, SwapErrorType } from 'lib/swapper/api'
+import { isEvmChainAdapter } from 'lib/utils'
+import { getWeb3InstanceByChainId } from 'lib/web3-instance'
+
+import { getLifi } from '../getLifi'
+
+export const getNetworkFeeCryptoBaseUnit = async ({
+  chainId,
+  lifiStep,
+}: {
+  chainId: ChainId
+  lifiStep: Step
+}) => {
+  const lifi = getLifi()
+  const adapter = getChainAdapterManager().get(chainId)
+
+  if (adapter === undefined || !isEvmChainAdapter(adapter)) {
+    throw new SwapError('[getNetworkFeeCryptoBaseUnit] invalid chain adapter', {
+      code: SwapErrorType.TRADE_QUOTE_FAILED,
+      details: { chainId },
+    })
+  }
+
+  const { transactionRequest } = await lifi.getStepTransaction(lifiStep)
+  const { value, from, to, data, gasLimit, gasPrice, maxFeePerGas } = transactionRequest ?? {}
+
+  if (value === undefined || from === undefined || to === undefined || data === undefined) {
+    throw new SwapError('[getNetworkFeeCryptoBaseUnit] getStepTransaction failed', {
+      code: SwapErrorType.TRADE_QUOTE_FAILED,
+    })
+  }
+
+  const gasFee = maxFeePerGas
+    ? bn(maxFeePerGas as string)
+    : bnOrZero(gasLimit as string).times(gasPrice as string)
+
+  // TEMP: this is necessary because infura currently cannot estimate gas for some lifi contract
+  // interactions, so instead of calling our existing stack which relies on infura we call the
+  // optimism gas oracle directly.
+  if (chainId === KnownChainIds.OptimismMainnet) {
+    const optimismGasOracleAddress = '0x420000000000000000000000000000000000000f'
```

Comment:
Nitpick, this is pretty well contained so not a big deal but do we want to abstract this as a `OPTIMISM_GAS_ORACLE_ADDRESS` constant outside this scope?